code/src/reducers/game.js

```diff
@@ -0,0 +1,70 @@
+import { createSlice } from '@reduxjs/toolkit';
+import { API_START, API_ACTION } from 'utils/urls'
+import { ui } from './ui';
+
+export const game = createSlice({
+  name: 'game',
+  initialState: {
+    currentPosition: null,
+    username: null,
+    history: []
+  },
+  reducers: {
+    setCurrentPosition: (store, action) => {
```

Comment:
Here you could add for the user and then in your actual game component declare it as const userName = useSelector((store) => store.gameState.showedUsername); to avoid winning straight a way